protocols/luci-proto-xfrm/htdocs/luci-static/resources/protocol/xfrm.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+'require uci';
+'require form';
+'require network';
+'require tools.widgets as widgets';
+
+return network.registerProtocol('xfrm', {
+	getI18n: function() {
+		return _('XFRM');
+	},
+
+	getIfname: function() {
+		return this._ubus('l3_device') || this.sid;
+	},
+
+	getOpkgPackage: function() {
+		return 'xfrm';
+	},
+
+	isFloating: function() {
+		return true;
+	},
+
+	isVirtual: function() {
+		return true;
+	},
+
+	getDevice: function() {
+		return null;
+	},
+
+	containsDevice: function(ifname) {
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+
+	renderFormOptions: function(s) {
+		var o, ss;
+
+		o = s.taboption('general', form.Value, 'ifid', _('Interface ID'), _('Required. XFRM interface ID to be used for SA.'));
+		o.datatype = 'integer';
+
+		o = s.taboption('general', widgets.NetworkSelect, 'tunlink', _('Required. Underlying interface.'));
+		o.exclude = s.section;
+		o.nocreate = true;
+
+		o = s.taboption('general', form.Value, 'mtu', _('MTU'), _('Required. Maximum Transmission Unit of the XFRM interface.'));
+		o.datatype = 'range(1280,1500)';
+		o.placeholder = '1492';
+		o.optional = true;
```

Comment:
Why is this required in your help message, if do you set this optional?

Reply:
Good catch, it was unintentional. From the xfrm package, it seems it should be optional.

> Is the range also correct here?

As I mentioned in the PR, I really have no idea what the range should be, would appreciate some guidance.